user/jwt.md

```diff
@@ -0,0 +1,155 @@
+2. Contact the third-party support and provide them with the encrypted token (echo the key in your test script), and link to the Travis job.
+
+## Third-Party Service Integration
+
+Third-party service needs to implement a new authentication method on the server side so that the JWT token is recognized and verified.
+
+### JWT Libraries
+
+[JWT.io](https://jwt.io) has a complete list of [supported languages](https://jwt.io/#libraries-io) and documentation on how everything works
+
+### Payload
+
+An example payload used to generate the JWT token:
+
+```javascript
+{
+  "iss": "Travis CI, GmbH",
+  "slug": "<SLUG>",
+  "pull-request": "<PR>",
+  "exp": <now+5400>,
+  "iat": <now>
+}
+```
+
+### Third Party Service Provider Code Sample 
+
+A code sample which illustrates how to add JWT token authentication to third party services.
+
+#### Python
+
+In this example we assume the authentication credentials (using environment variables
+e.g. `SERVICE_USERNAME` + `SERVICE_ACCESS_KEY`) of a RESTful API will be sent as HTTP BASIC AUTH header:
+
+```
+Authorization: Basic am9obmRvZTpleUowZVhBaU9pSktWMVFpTENKaGJHY2lPaUpJVXpJMU5pSjkuZXlKcGMzTWlPaUow\nY21GMmFYTXRZMmt1YjNKbklpd2ljMngxWnlJNkluUnlZWFpwY3kxamFTOTBjbUYyYVhNdFkya2lM\nQ0p3ZFd4c0xYSmxjWFZsYzNRaU9pSWlMQ0psZUhBaU9qVTBNREFzSW1saGRDSTZNSDAuc29RSmdI\nUjZjR05yOUxqX042eUwyTms1U1F1Zy1oWEdVUGVuSnkxUVRWYw==
+```
+
+The HTTP BASIC AUTH header's payload is base64 encoded which will decode to string as follows.
+
+```
+johndoe:eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJ0cmF2aXMtY2kub3JnIiwic2x1ZyI6InRyYXZpcy1jaS90cmF2aXMtY2kiLCJwdWxsLXJlcXVlc3QiOiIiLCJleHAiOjU0MDAsImlhdCI6MH0.soQJgHR6cGNr9Lj_N6yL2Nk5SQug-hXGUPenJy1QTVc
+```
+
+The colon separated string contains the username before the colon and the JWT
+token after the colon. The username is used to retrieve the user object from
+the user database. Below is a function which is executed against the user
+object and the token to validate them for authentication. Please note that the
+code is deliberately agnostic to what value the access_key contains. It doesn't
+matter whether a JWT token or an access key is passed into the function.
+However, service providers will have to add the JWT auth attempt to an already
+existing authentication mechanism.
+
+```python
+import jwt
+
+def authenticate(user, access_key):
+    """
+    user: db object representing user retrieved based on username from HTTP BASIC AUTH
+    access_key: access key or JWT token signed using access key (shared secret)
+    returns True when authenication validation passed, otherwise False
+    """
+    # primary auth method
+    if user['access_key'] == access_key:
+        return True
+
+    # secondary auth attempt using JWT method
+    try:
+        return bool(jwt.decode(access_key, user['access_key']))
+    except (jwt.DecodeError, jwt.ExpiredSignature):
+        return False
+```
+
+## List of Third-Party Services Integrated with the JWT Addon
+
+### Sauce Labs
```

Comment:
I've fixed up the example (it shouldn't mention sauce connect) and added a link to sauce connect for more details.